### ISSUE TITLE
[RMP] Quick-start for ranking models training pipeline

### ISSUE THREAD

**Issue by gabrielspmoreira** [NVIDIA-Merlin/Merlin#827]
## Problem:
Merlin provides documentation and a number of example notebooks on how to use tools like NVTabular, Dataloader and Merlin Models. In order to build a pipeline for training and evaluation purposes, a Data Scientist needs to analyze that material, copy-and-paste code snippets demonstrating the API and glue that code together to implement scripts for experimentation and benchmarking.
It might also not be clear to the users the advanced API options featured by Merlin Models that can be mapped as a hyperparameter, and potentially improve models accuracy.

## Goal:
This RMP provides a Quick-start for building ranking models training pipelines. 
It is composed by:

### Template scripts
- Generic template script for preprocessing
- Generic template script for training ranking models, exposing the main hyperparameters for ranking models .
It includes support to ranking models like DLRM, DCN-v2, DeepFM, Wide&Deep and multi-task learning (MTL) like MMOE, CGC and PLE.

### Documentation

- Documentation of the scripts command line arguments
- Documentation of best practices learned from our experimentation:
    - Hyperparameter tuning: search space, most important hyperparameters and best hparams for TenRec public dataset for STL and MTL models
    - Intuitions of API options (building blocks, arguments) that can improve models accuracy

## Constraints:
- Preprocessing - The pre-processing template notebook will perform some basic feature encoding for categorical (e.g. categorify) and continuous variables (e.g. standardization). The customer can expand the template with advanced preprocessing ops demonstrated in our examples.
- Training - The training and evaluation script for Merlin Models should be totally configurable, taking as input the parquet files and schema, and a number of hyperparameters exposed via command line arguments. The output of this script should be the evaluation metrics, logged as a CSV file and also to Weights&Biases.

## Starting Point:
The ranking training scripts we have developed for the [MTL research project](https://github.com/NVIDIA-Merlin/research/tree/main/MTL).

### Ranking Models (Steps 4-7)
PR: https://github.com/NVIDIA-Merlin/models/pull/988

- [x] https://github.com/NVIDIA-Merlin/models/issues/985
- [x] https://github.com/NVIDIA-Merlin/models/issues/914
- [x] NVIDIA-Merlin/models#986
- [x] NVIDIA-Merlin/models#987

#### Tenrec dataset experiments
- [ ] NVIDIA-Merlin/Merlin#680
- [ ] NVIDIA-Merlin/Merlin#681

#### Research & Documentation
- [ ] NVIDIA-Merlin/Merlin#666

#### Testing
- [ ] NVIDIA-Merlin/Merlin#667

**Comment by rnyak** [comment]
The task https://github.com/NVIDIA-Merlin/Merlin/issues/919 is handled by https://github.com/NVIDIA-Merlin/Merlin/pull/966.
